components/bpel/org.wso2.carbon.bpel/src/main/java/org/wso2/carbon/bpel/core/ode/integration/mgt/services/InstanceManagementServiceSkeleton.java

```diff
@@ -1452,14 +1452,19 @@ private ActivityInfoType fillActivityInfo(ActivityInfoType activity, TActivityIn
      * @param iid instance id
      * @throws IllegalAccessException if instance doesn't belong to the current tenant
      * @throws ProcessingException    if there a error getting instance data
+     * @throws InstanceManagementException if the processId does not exist
      */
     private void isOperationIsValidForTheCurrentTenant(final long iid) throws IllegalAccessException,
-            ProcessingException {
+            ProcessingException, InstanceManagementException {
         QName processId = getProcess(iid);
         TenantProcessStoreImpl processStore = getTenantProcessForCurrentSession();
-        if (!processStore.containsProcess(processId)) {
-            log.error("Trying to invoke a illegal operation. Instance ID:" + iid);
-            throw new IllegalAccessException("Operation is not permitted.");
+        if (processId != null) {
+            if (!processStore.containsProcess(processId)) {
```

Comment:
Can merge these two if blocks to one

Reply:
If we merge the two if blocks we should null check processId variable again and wont it be a performance issue to check the same variable in two if blocks?

Reply:
+1, yeah you are correct